version: "3.8"
services:
  app:
    build: .
    volumes:
      - .:/django
    ports:
      - 8000:8000
    image: app:django
    container_name: django_container
    command: python manage.py runserver 0.0.0.0:8000
    depends_on:
      - db 
  db:
    container_name: postgres_db
    image: postgres
    restart: always 
    volumes:
      - ./data/db:/var/lib/postgresql/data
    environment:
      - POSTGRES_DB=postgres
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
    ports:
      - "5432:5432"
    networks:
      app_net:
        ipv4_address: 192.168.0.2  

    
  pgadmin:
    container_name: pgadmin4
    image: dpage/pgadmin4
    restart: always

    volumes:
      - ./data/pgadmin-data:/var/lib/pgadmin

    environment:
      PGADMIN_DEFAULT_EMAIL: root@root.com
      PGADMIN_DEFAULT_PASSWORD: root
      PGADMIN_LISTEN_PORT: 80
    ports:
      - "5050:80"
    networks:
      app_net:
        ipv4_address: 192.168.0.3

networks:
  app_net:
    ipam:
      driver: default
      config:
        - subnet: "192.168.0.0/24"
          gateway: 192.168.0.1        